integration_tests/tests/conftest.py

```diff
@@ -2,17 +2,61 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+
 from datetime import timedelta
+from multiprocessing import Process
+from time import sleep
 
 import pytest
 import os
 
 import requests
-from .utils import make_token
 
 import flowmachine
+
+
 from flowmachine.core import Connection, Query
 from flowmachine.core.cache import reset_cache
+import flowmachine.core.server.server
+import quart.flask_patch
+
+
+@pytest.fixture(scope="session", autouse=True)
+def flowmachine_server():
+    """
+    Starts a flowmachine server in a separate process for the tests to talk to.
+    """
+    fm_thread = Process(target=flowmachine.core.server.server.main)
+    fm_thread.start()
+    yield
+    fm_thread.terminate()
+    sleep(2)  # Wait a moment to make sure coverage of subprocess finishes being written
+
+
+@pytest.fixture(scope="session", autouse=True)
+def flowapi_server():
+    """
+    Starts a FlowAPI server in a separate process for the tests to talk to.
+    """
+    import hypercorn.__main__
+
+    import os
+    import sys
+
+    sys.path.insert(
+        0,
+        os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "..", "flowapi")),
+    )
+
+    api_thread = Process(
+        target=hypercorn.__main__.main,
+        args=(["--bind", "0.0.0.0:9090", "app.main:create_app()"],),
```

Comment:
Is this hard-coded port going to conflict with any other instances of flowapi that are also running? How about reading the port from an environment variable (e.g. `FLOWKIT_INTEGRATION_TESTS_FLOWAPI_PORT` or similar)? (See the `zmq_port` fixture below.)

Similarly for the `flowmachine_server` above - this will start on port 5555 and might conflict with an existing flowmachine server, right?

Reply:
Good point. I think actually the right thing to do here is to not specify them and use something similar to pytest-asyncio's [unused_tcp_port](https://github.com/pytest-dev/pytest-asyncio#unused_tcp_port) fixture to specify + try to bind on [port 0](https://www.dnorth.net/2012/03/17/the-port-0-trick/) in the actual code. But I'm gonna spin that off into an issue I think.